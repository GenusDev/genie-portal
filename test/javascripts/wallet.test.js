
import React from 'react';
import Wallet from 'Entities/navbar/user_dropdown/wallet/wallet';
import { shallow, mount } from 'enzyme';

function setUp(){
 const props = {
   fetchProjects: jest.fn(() => {
     return Promise.resolve();
   })
 };
 const eWrapper = shallow(<Wallet {...props}/>);
 const mountWrapper = mount(<Wallet {...props}/>);
 return {
   props,
   eWrapper,
   mountWrapper
 };
}

describe('Wallet information', () => {
  it('should render without throwing an error', () => {
    const { eWrapper } = setUp()
    expect(eWrapper).toMatchSnapshot();
  });
  it('expects props', () => {
    const { eWrapper } = setUp()
    expect(eWrapper.props).toBeTruthy();
  });
})
