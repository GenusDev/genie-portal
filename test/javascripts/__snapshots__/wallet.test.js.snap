// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet information should render without throwing an error 1`] = `
<form
  className="wallet-form"
>
  <div
    className="wallet current-wallet"
  >
    <p
      className="wallet-label current-wallet-label"
    >
      current wallet
    </p>
    <div
      className="wallet-id current-wallet-id"
    />
  </div>
  <div
    className="wallet integrate-wallet"
  >
    <p
      className="wallet-label integrate wallet-label"
    >
      integrate new wallet
    </p>
    <div
      className="input-container"
    >
      <input
        className="wallet-id integrate-wallet-input"
        onChange={[Function]}
        placeholder="################"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        <img
          className="wallet-submit"
          src="https://s3.amazonaws.com/genie-portal-dev/static/wallet.svg"
        />
      </button>
    </div>
  </div>
</form>
`;
